nonos/api/from_simulation.py

```diff
@@ -0,0 +1,672 @@
+}
+
+
+class Parameters:
+    def __init__(
+        self,
+        *,
+        inifile: str = "",
+        code: str = "",
+        directory: str = "",
+    ):
+        self.directory = directory
+        self.paramfile = inifile
+        self.code = code
+        if (self.code, self.paramfile).count("") == 0:
+            found = Path(self.directory).joinpath(self.paramfile).is_file()
+            if not found:
+                raise FileNotFoundError(f"{self.paramfile} not found.")
+        elif (self.code, self.paramfile).count("") == 2:
+            found_dict = {
+                paramfile: Path(self.directory).joinpath(paramfile).is_file()
+                for paramfile in POSSIBLE_INIFILES
+            }
+            nfound = sum(list(found_dict.values()))
+            if nfound == 0:
+                raise FileNotFoundError(
+                    f"{', '.join(POSSIBLE_INIFILES.keys())} not found."
+                )
+            elif nfound > 1:
+                raise RuntimeError("found more than one possible ini file.")
+            self.paramfile = list(POSSIBLE_INIFILES.keys())[
+                list(found_dict.values()).index(True)
+            ]
+            self.code = POSSIBLE_INIFILES[self.paramfile]
+        elif (self.code, self.paramfile).count("") == 1:
+            raise ValueError("both inifile and code have to be given.")
+
+    def loadIniFile(self):
+        self.inifile = inifix.load(path.join(self.directory, self.paramfile))
+        if self.code == "idefix":
+            self.vtk = self.inifile["Output"]["vtk"]
+        elif self.code == "pluto":
+            self.vtk = self.inifile["Static Grid Output"]["vtk"][0]
+        elif self.code == "fargo3d":
+            self.vtk = self.inifile["NINTERM"] * self.inifile["DT"]
+
+    def loadPlanetFile(self, *, planet_number: int = 0):
+        planet_file = f"planet{planet_number}.dat"
+
+        if self.code in ("idefix", "fargo3d"):
+            if Path(self.directory).joinpath(planet_file).is_file():
+                columns = np.loadtxt(path.join(self.directory, planet_file)).T
+                self.qpl = columns[7]
+                self.dpl = np.sqrt(np.sum(columns[1:4] ** 2, axis=0))
+                self.xpl = columns[1]
+                self.ypl = columns[2]
+                self.tpl = columns[8]
+            else:
+                raise FileNotFoundError(f"{planet_file} not found")
+        else:
+            raise FileNotFoundError(
+                f"{planet_file} not found for {self.code}. For now, you can't rotate the grid with the planet."
+            )
+
+    def countSimuFiles(self):
+        if self.code == "fargo3d":
+            self.data_files = [
+                fn
+                for fn in glob.glob1(self.directory, "gasdens*.dat")
+                if re.match(r"gasdens\d+.dat", fn)
+            ]
+        elif self.code in ("idefix", "pluto"):
+            self.data_files = list(glob.glob1(self.directory, "data.*.vtk"))
+
```

Comment:
This is a general remark: be careful how your methods/functions behave if `self.code` is a currently unsupported value. It shouldn't affect users, but it'll make bugs much easier to find if we ever want to add support for other formats.
```suggestion
            self.data_files = list(glob.glob1(self.directory, "data.*.vtk"))
        else:
            raise RuntimeError("Unknown file format")
```